watsong/util.py

```diff
@@ -0,0 +1,21 @@
+from typing import Iterator, List
+
+from .structures import T
+
+
+def chunks(elements: Iterator[T], n: int) -> Iterator[List[T]]:
+    """
+    Yield successive n-sized chunks from elements.
+    """
+    at_end = False
+
+    while not at_end:
+        chunk = []
+        for i in range(n):
+            try:
+                chunk.append(next(elements))
+            except StopIteration:
+                at_end = True
+                break
```

Comment:
It would be a bit more pythonic if you remove the at_end keyword and instead just return instead of break on exception thrown

Reply:
I tried it but I can't figure out how to create a generator expression correctly. I want it to do:

```python
chunks(iter(range(5)), 2) # => [[0, 1], [2, 3], [4]]
```

Any ideas?

Reply:
```python
while True:
    chunk = []
    for i in range(n):
        try:
            chunk.append(next(elements))
        except StopIteration:
            yield chunk
            return

    yield chunk
```
Works!